rskj-core/src/main/java/co/rsk/peg/LegacyErpFederation.java

```diff
@@ -83,6 +83,17 @@ private void validateRedeemScript() {
             logger.debug("[validateRedeemScript] {}", message);
             throw new FederationCreationException(message);
         }
+
+        // we have to check if the size of every script inside the scriptSig is not above the maximum
+        // this scriptSig contains the signatures, the redeem script and some other bytes
+        // so it is ok to just check the redeem script size
+
+        int bytesFromRedeemScript = getRedeemScript().getProgram().length;
+
+        if (bytesFromRedeemScript > Standardness.MAX_SCRIPT_ELEMENT_SIZE) {
+            String message = "Unable to create LegacyErpFederation. The redeem script size is above the maximum allowed.";
+            throw new FederationCreationException(message);
+        }
```

Comment:
We Are Doing This Same Validation For All Federation Types.. Maybe Move This To A Utils File Or The Base Class?

Reply:
I agree on moving it to a util file where you have the fed validation methods because you may have some federations that have validations in commons, thus you don't need to repeat the same method if you have it in a util class.

Reply:
the reason why it is not in the base class is because this validation is no accurate for all federations but for legacy ones.
actually, its purpose is to have the size of the scriptsig validated, not the redeem script itself. but in legacy we have the redeemscript inside the scriptsig.
segwit (compatible or native) will be another erp federation that has a fixed size for the scriptsig, so this validation wont be right for it.

but i think that moving it into a utils file is good though, thanks!